src/main/java/org/terasology/genome/genomeMap/GeneIndexGenomeMap.java

```diff
@@ -16,37 +16,54 @@
 package org.terasology.genome.genomeMap;
 
 import com.google.common.base.Function;
+import org.terasology.genome.breed.BreedingAlgorithm;
+import org.terasology.genome.breed.DiploidBreedingAlgorithm;
+import org.terasology.genome.breed.MonoploidBreedingAlgorithm;
 
-import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Map;
 
 /**
  * An implementation of GenomeMap.
  */
 public class GeneIndexGenomeMap implements GenomeMap {
-    private Map<String, GenePropertyDefinition> propertyDefinitionMap = new HashMap<>();
+    public Map<String, GenePropertyDefinition> propertyDefinitionMap = new LinkedHashMap<>();
 
     /**
      * Add a new named property to the genome map.
      *
-     * @param propertyName             The name of the property to be added
-     * @param geneIndices              The indices of the genes which the property depends
-     * @param type                     The type of the property
+     * @param propertyName The name of the property to be added
+     * @param geneIndices The indices of the genes which the property depends
+     * @param type The type of the property
+     * @param breedingAlgorithm The breeding algorithm of the property
      * @param geneStringTransformation A function to transform the gene into a property value
-     * @param <T>                      The class used for the property value
+     * @param <T> The class used for the property value
      */
-    public <T> void addProperty(String propertyName, int[] geneIndices, Class<T> type, Function<String, T> geneStringTransformation) {
-        propertyDefinitionMap.put(propertyName, new GenePropertyDefinition<T>(geneIndices, type, geneStringTransformation));
+    public <T> void addProperty(String propertyName, int[] geneIndices, Class<T> type,
+                                BreedingAlgorithm breedingAlgorithm,
+                                Function<String, T> geneStringTransformation) {
+        propertyDefinitionMap.put(propertyName, new GenePropertyDefinition<T>(geneIndices, type, breedingAlgorithm,
+                geneStringTransformation));
     }
 
+    /**
+     * Get the breeding algorithm of the specified property.
+     *
+     * @param propertyName The name of the property to be added
+     */
+    public BreedingAlgorithm getPropertyBreedingAlgorithm(String propertyName) {
+        return propertyDefinitionMap.get(propertyName).breedingAlgorithm;
```

Comment:
This will easily throw a NPE if the property definitions does not contain the given property name. Consider changing the return type of this method to `Optional<BreedingAlgorithm>` and make the implementation null-safe, e.g., 

```java
return Optional.ofNullable(propertyDefinitionMap.get(propertyName))
            .map(GenePropertyDefinition::breedingAlgorithm);
```

Reply:
This method is actually not used as it is sufficient to use the propertyDefinitionMap for the breeding algorithm of the traits, therefore I have removed it. But this is a fair point and I will keep it in mind for future work. For knowledge, is it preferred to use the Optional class for such a scenario or will a local variable with null check suffice?